scripts/pancancer.py

```diff
@@ -0,0 +1,341 @@
+    log = logging.getLogger(__name__)
+    def __init__(self,Coordinator,args):
+        self.service_name='Coordinator'
+
+###
+
+class Provisioner(DaemonCommand):
+    "The Provisioner will launcher worker VMs as needed based on the workflow orders sent to the system."
+    log = logging.getLogger(__name__)
+    def __init__(self,Provisioner,args):
+        self.service_name='Provisioner'
+
+###
+
+class Generator(Command):
+    "This Generator will generate new job orders."
+    log = logging.getLogger(__name__)
+    def get_parser(self,prog_name):
+       parser = super(Generator,self).get_parser(prog_name)
+       parser.add_argument('--workflow',dest='workflow_name',help='The name of the workflow for which you would like to generate jobs.',required=True)
+       return parser
+
+    def take_action(self, parsed_args):
+        workflow_name=vars(parsed_args)['workflow_name']
+        self.log.debug('workflow_name: %s',workflow_name)
+        if not (workflow_name in WorkflowLister.get_workflow_names()):
+            print('Oh, I\'m SO sorry, but '+workflow_name+' is not the name of an available workflow.\nPlease use the command \'workflows list\' to see the list of currently available workflows.')
+        else:
+            workflow_details = WorkflowLister.get_workflow_details(workflow_name)
+            generator_cmd = 'Generator --workflow-name '+workflow_name+' --workflow-version '+workflow_details['http_workflow']['version']+' --workflow-path '+'/workflows/'+workflow_details['full_name']+' --ini-dir '+'/home/ubuntu/ini-dir --config /home/ubuntu/arch3/config/masterConfig.ini'
+            self.log.debug(generator_cmd)
+            # Before generating the job, we have to update params.json with the workflow/container info about the requested workflow.
+            paramsData=''
+            with open('/home/ubuntu/params.json','r') as params_file:
+                paramsData = json.load(params_file)
+                if 'http_workflow' in workflow_details:
+                    paramsData['http_workflows'] = workflow_details['http_workflow']
+                if 's3_workflows' in workflow_details:
+                    paramsData['s3_workflows'] = workflow_details['s3_workflows']
+                if 'containers' in workflow_details:
+                    paramsData['containers'] = workflow_details['containers']
+                if 's3_containers' in workflow_details:
+                    paramsData['s3_containers'] = workflow_details['s3_containers']
+                if 'http_containers' in workflow_details:
+                    paramsData['http_containers'] = workflow_details['http_containers']
+
+                paramsData['lvm_device_whitelist'] = workflow_details['lvm_devices']
+                paramsData['workflow_name'] = workflow_name
+
+            # Now write the params.json file.
+            with open('/home/ubuntu/params.json','w+') as params_file:
```

Comment:
Well, I want to load the json data from the file, tweak a few parts based on what the user selected, then write back to a file. So I read the whole file, tweaked the data, then overwrote the file with the new data. I couldn't find an easy way to edit the json data in the file without overwriting (not overwriting produced duplicates of the main data structure which caused errors the next time I tried to read the file).